app/models/evaluation.rb

```diff
@@ -44,6 +44,8 @@ class Evaluation < ApplicationRecord
 
   before_save :ensure_all_scores_exist
   before_save :calculate_total_score
+  after_destroy :update_submission_evaluation_status
+  after_save :update_submission_evaluation_status, if: -> { saved_change_to_completed_at? || saved_change_to_id? }
```

Comment:
I have rarely seen `saved_change_to_id?` but it seems like it could work. usually I would just break this into separate callbacks on _create and _update
```suggestion
  after_create :update_submission_evaluation_status
  after_update :update_submission_evaluation_status, if: -> { saved_change_to_completed_at? }
```